### ISSUE TITLE
Buildbot builds all failing

### ISSUE THREAD

**Issue by khuck** [STEllAR-GROUP/phylanx#770]
Commits merged on Feb 4 have broken the build:
http://ktau.nic.uoregon.edu:8020/#/builders/8/builds/385

```
[ 25%] Linking CXX shared library ../lib/libhpx_phylanx.so
CMakeFiles/phylanx_component.dir/execution_tree/primitives/slice_node_data.cpp.o: In function `phylanx::ir::node_data<unsigned char> phylanx::execution_tree::slice_assign<unsigned char>(phylanx::ir::node_data<unsigned char>&&, phylanx::execution_tree::primitive_argument_type const&, phylanx::ir::node_data<unsigned char>&&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&)':
slice_node_data.cpp:(.text._ZN7phylanx14execution_tree12slice_assignIhEENS_2ir9node_dataIT_EEOS5_RKNS0_23primitive_argument_typeES6_RKNSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEESH_[_ZN7phylanx14execution_tree12slice_assignIhEENS_2ir9node_dataIT_EEOS5_RKNS0_23primitive_argument_typeES6_RKNSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEESH_]+0xa0): undefined reference to `void phylanx::execution_tree::extract_value_scalar<unsigned char>(phylanx::ir::node_data<unsigned char>::storage0d_type&, phylanx::ir::node_data<unsigned char>&&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&)'
CMakeFiles/phylanx_component.dir/execution_tree/primitives/slice_node_data.cpp.o: In function `phylanx::ir::node_data<double> phylanx::execution_tree::slice_assign<double>(phylanx::ir::node_data<double>&&, phylanx::execution_tree::primitive_argument_type const&, phylanx::ir::node_data<double>&&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&)':
slice_node_data.cpp:(.text._ZN7phylanx14execution_tree12slice_assignIdEENS_2ir9node_dataIT_EEOS5_RKNS0_23primitive_argument_typeES6_RKNSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEESH_[_ZN7phylanx14execution_tree12slice_assignIdEENS_2ir9node_dataIT_EEOS5_RKNS0_23primitive_argument_typeES6_RKNSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEESH_]+0xa0): undefined reference to `void phylanx::execution_tree::extract_value_scalar<double>(phylanx::ir::node_data<double>::storage0d_type&, phylanx::ir::node_data<double>&&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&)'
CMakeFiles/phylanx_component.dir/execution_tree/primitives/slice_node_data.cpp.o: In function `phylanx::ir::node_data<long> phylanx::execution_tree::slice_assign<long>(phylanx::ir::node_data<long>&&, phylanx::execution_tree::primitive_argument_type const&, phylanx::ir::node_data<long>&&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&)':
slice_node_data.cpp:(.text._ZN7phylanx14execution_tree12slice_assignIlEENS_2ir9node_dataIT_EEOS5_RKNS0_23primitive_argument_typeES6_RKNSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEESH_[_ZN7phylanx14execution_tree12slice_assignIlEENS_2ir9node_dataIT_EEOS5_RKNS0_23primitive_argument_typeES6_RKNSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEESH_]+0xa0): undefined reference to `void phylanx::execution_tree::extract_value_scalar<long>(phylanx::ir::node_data<long>::storage0d_type&, phylanx::ir::node_data<long>&&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&)'
collect2: error: ld returned 1 exit status
make[2]: *** [lib/libhpx_phylanx.so.0.0.1] Error 1
make[1]: *** [src/CMakeFiles/phylanx_component.dir/all] Error 2
make: *** [all] Error 2
```

**Comment by hkaiser** [comment]
@khuck strange, CircleCI passes just fine :/

**Comment by khuck** [comment]
@hkaiser of course it does.  :)   "it works for me"...

It seems to be a side effect of an HPX change - if I go back to HPX 09f08b88d513ead41646704215b64661cffd3ccd , phylanx compiles fine... ?  I'll start bisecting and see what the offending commit was.

**Comment by khuck** [comment]
@hkaiser And the guilty party is HPX merge request #3661 https://github.com/STEllAR-GROUP/hpx/pull/3661.  Not sure why.  Phylanx builds fine with HPX with merge request #3660, but fails after #3661.

**Comment by sithhell** [comment]
Ha, my bad. I guess you're using clang3? There's a bug with the thread_local implementation leading to that error...

**Comment by hkaiser** [comment]
@sithhell the missing functions have nothing in common with the thread_local stuff. Could you elaborate on where this connection is, please?

**Comment by sithhell** [comment]
@hkaiser you are right ... I was referring to the rostam buildbot failures on clang3... totally different....

**Comment by hkaiser** [comment]
This was fixed by merging #772
